server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -222,24 +219,20 @@ public void stop()
 
   @Provides
   @Singleton
-  public JettyMonitor getJettyMonitor(Properties props)
+  public JettyMonitor getJettyMonitor(
+      DataSourceTaskIdHolder dataSourceTaskIdHolder
+  )
   {
-    return new JettyMonitor(props);
+    return new JettyMonitor(dataSourceTaskIdHolder.getDataSource(), dataSourceTaskIdHolder.getTaskId());
```

Comment:
@fjy you and @xvrl  need to reconcile if pulling from task or pulling from properties is better. I'm pretty impartial.